FEAT-#4897: Drop python 3.6 support.

mvashishtha opened the pull request:
Signed-off-by: mvashishtha <mahesh@ponder.io><!--
Thank you for your contribution!
Please review the contributing docs: https://modin.readthedocs.io/en/latest/development/contributing.html
if you have questions about contributing.
-->

## What do these changes do?

<!-- Please give a short brief about these changes. -->

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` <!-- you can amend your commit with a signature via `git commit -amend -s` -->
- [x] Resolves #4897
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date <!-- if you have added, renamed or removed files or directories please update the documentation accordingly -->

mvashishtha commented:
@lgtm-com I think all the errors you mention were already present before we added compatibility mode. I don't think we should fix them in this PR.

vnlitvinov commented on modin/pandas/general.py:
```diff
@@ -318,6 +318,42 @@ def merge_asof(
+    margins_name="All",
+    observed=False,
+    sort=True,
+):  # noqa: PR01, RT01, D200
+    """
+    Create a spreadsheet-style pivot table as a DataFrame.
+    """
```
do we need these `noqa`-s and docstring given we inherit the doc?

vnlitvinov commented on modin/pandas/io.py:
```diff
@@ -19,38 +19,385 @@
+        val.name for val in inspect.signature(pandas.read_csv).parameters.values()
+    }
+    _, _, _, f_locals = inspect.getargvalues(inspect.currentframe())
+    # mangle_dupe_cols has no effect starting in pandas 1.5. Exclude it from
+    # kwargs so pandas doesn't spuriously warn people not to use it.
+    f_locals.pop("mangle_dupe_cols", None)
```
can we drop that?

mvashishtha replied:
No, not as the code is written-- we have `{"mangle_dupe_cols": True}` in the dict of `f_locals`, so we explicitly pass it into pandas `read_csv` unless we drop it.

vnlitvinov commented on modin/pandas/io.py:
```diff
@@ -19,38 +19,385 @@
+    _, _, _, f_locals = inspect.getargvalues(inspect.currentframe())
+    if f_locals.get("sep", sep) is False or f_locals.get("sep", sep) is no_default:
+        f_locals["sep"] = "\t"
+    # mangle_dupe_cols has no effect starting in pandas 1.5. Exclude it from
+    # kwargs so pandas doesn't spuriously warn people not to use it.
+    f_locals.pop("mangle_dupe_cols", None)
```
maybe drop?

mvashishtha replied:
ditto from other mangle_dupe_cols comment

vnlitvinov commented on modin/pandas/series.py:
```diff
@@ -129,7 +129,6 @@ def __init__(
             )._query_compiler
         self._query_compiler = query_compiler.columnarize()
         if name is not None:
-            self._query_compiler = self._query_compiler
```
lol

vnlitvinov commented on modin/pandas/series.py:
```diff
@@ -612,7 +611,9 @@ def error_raiser(msg, exception):
-    def _apply(self, func, convert_dtype, args, **kwargs):  # noqa: PR01, RT01, D200
+    def apply(
+        self, func, convert_dtype=True, args=(), **kwargs
+    ):  # noqa: PR01, RT01, D200
```
I don't think we need the `noqa`, docstring should be inherited

mvashishtha replied:
I get a doc_checker error if I remove this noqa.

mvashishtha commented on modin/pandas/dataframe.py:
```diff
@@ -375,22 +380,28 @@ def add_suffix(self, suffix):  # noqa: PR01, RT01, D200
+    ):  # noqa: PR01, RT01, D200
         """
         Apply a function to a ``DataFrame`` elementwise.
         """
```
We can get rid of the noqa and docstring for `applymap`, but not for many of the other methods I added noqa for. I don't understand the pattern.

mvashishtha commented on modin/pandas/dataframe.py:
```diff
@@ -2242,6 +2323,32 @@ def to_html(
+            "partition_cols": partition_cols,
+            "storage_options": storage_options,
+        }
+        new_query_compiler = self._query_compiler
```
Nope, I got rid of `config`.

anmyachev commented on docs/supported_apis/older_pandas_compat.rst:
```diff
@@ -1,37 +0,0 @@
-===================================
-Pandas backwards compatibility mode
```
Maybe we should leave a mention of this mode, indicating the latest release, where is it available?

mvashishtha replied:
Done.

anmyachev commented on modin/core/dataframe/algebra/default2pandas/rolling.py:
```diff
@@ -39,8 +39,7 @@ def build_rolling(cls, func):
 
         def fn(df, rolling_args, *args, **kwargs):
             """Create rolling window for the passed frame and execute specified `func` on it."""
-            rolling_args, rolling_kw = rolling_args
-            roller = df.rolling(*rolling_args, **rolling_kw)
```
It's not obvious. Maybe create a issue for that?

anmyachev replied:
I'm not sure, it just looks weird.

anmyachev commented on modin/core/io/io.py:
```diff
@@ -28,16 +28,22 @@
-from modin._compat.core.pandas_common import pandas_to_csv
+_doc_default_io_method = """
+{summary} using pandas.
+For parameters description please refer to pandas API.
+Returns
```
?
```suggestion
For parameters description please refer to pandas API.

Returns
```

anmyachev commented on modin/pandas/series.py:
```diff
@@ -1544,8 +1567,13 @@ def repeat(self, repeats, axis=None):  # noqa: PR01, RT01, D200
 
-    def _reset_index(
-        self, level, drop, name, inplace, allow_duplicates
+    def reset_index(
```
@mvashishtha why we still don't use `@_inherit_docstrings` for all methods in Series? Maybe you know.

anmyachev replied:
In fact, this decorator applies to the entire class and therefore to all public methods. I was confused by the fact that this function has a redefined docstring. I guess we need to remove them in a separate pull request.

anmyachev replied:
https://github.com/modin-project/modin/issues/5286

anmyachev commented on modin/pandas/io.py:
```diff
@@ -66,6 +413,55 @@ def read_clipboard(sep=r"\s+", **kwargs):  # pragma: no cover  # noqa: PR01, RT0
+@enable_logging
+def read_excel(
+    io,
+    sheet_name: "str | int | list[IntStrT] | None" = 0,
+    header: "int | Sequence[int] | None" = 0,
+    names=None,
+    index_col: "int | Sequence[int] | None" = None,
+    usecols=None,
+    squeeze: "bool | None" = None,
+    dtype: "DtypeArg | None" = None,
+    engine: "Literal[('xlrd', 'openpyxl', 'odf', 'pyxlsb')] | None" = None,
+    converters=None,
+    true_values: "Iterable[Hashable] | None" = None,
+    false_values: "Iterable[Hashable] | None" = None,
+    skiprows: "Sequence[int] | int | Callable[[int], object] | None" = None,
+    nrows: "int | None" = None,
+    na_values=None,
+    keep_default_na: "bool" = True,
+    na_filter: "bool" = True,
+    verbose: "bool" = False,
+    parse_dates=False,
+    date_parser=None,
+    thousands: "str | None" = None,
+    decimal: "str" = ".",
+    comment: "str | None" = None,
+    skipfooter: "int" = 0,
+    convert_float: "bool | None" = None,
+    mangle_dupe_cols: "bool" = True,
+    storage_options: "StorageOptions" = None,
+) -> "DataFrame | dict[IntStrT, DataFrame]":
```
```suggestion
    sheet_name: str | int | list[IntStrT] | None = 0,
    header: int | Sequence[int] | None = 0,
    names: list[str] | None = None,
    index_col: int | Sequence[int] | None = None,
    usecols: int
    | str
    | Sequence[int]
    | Sequence[str]
    | Callable[[str], bool]
    | None = None,
    squeeze: bool | None = None,
    dtype: DtypeArg | None = None,
    engine: Literal[("xlrd", "openpyxl", "odf", "pyxlsb")] | None = None,
    converters: dict[str, Callable] | dict[int, Callable] | None = None,
    true_values: Iterable[Hashable] | None = None,
    false_values: Iterable[Hashable] | None = None,
    skiprows: Sequence[int] | int | Callable[[int], object] | None = None,
    nrows: int | None = None,
    na_values=None,
    keep_default_na: bool = True,
    na_filter: bool = True,
    verbose: bool = False,
    parse_dates: list | dict | bool = False,
    date_parser: Callable | None = None,
    thousands: str | None = None,
    decimal: str = ".",
    comment: str | None = None,
    skipfooter: int = 0,
    convert_float: bool | None = None,
    mangle_dupe_cols: bool = True,
    storage_options: StorageOptions = None,
) -> DataFrame | dict[IntStrT, DataFrame]:
```

anmyachev commented on modin/pandas/io.py:
```diff
@@ -176,6 +657,49 @@ def read_sql_table(
+    compression: "CompressionOptions" = "infer",
+    protocol: int = pickle.HIGHEST_PROTOCOL,
+    storage_options: "StorageOptions" = None,
+):
```
```suggestion
) -> None:
```

anmyachev commented on modin/pandas/io.py:
```diff
@@ -176,6 +657,49 @@ def read_sql_table(
+def read_sql_query(
+    sql,
+    con,
+    index_col=None,
+    coerce_float=True,
+    params=None,
+    parse_dates=None,
+    chunksize=None,
+    dtype=None,
+):
```
```suggestion
    index_col: str | list[str] | None = None,
    coerce_float: bool = True,
    params: list[str] | dict[str, str] | None = None,
    parse_dates: list[str] | dict[str, str] | None = None,
    chunksize: int | None = None,
    dtype: DtypeArg | None = None,
) -> DataFrame | Iterator[DataFrame]:
```

anmyachev commented on modin/pandas/io.py:
```diff
@@ -93,6 +489,91 @@ def read_hdf(
+@enable_logging
+def read_pickle(
+    filepath_or_buffer,
+    compression: Optional[str] = "infer",
+    storage_options: "StorageOptions" = None,
```
```suggestion
    compression: CompressionOptions = "infer",
    storage_options: StorageOptions = None,
```

anmyachev commented on modin/pandas/io.py:
```diff
@@ -93,6 +489,91 @@ def read_hdf(
+@enable_logging
+def read_sas(
+    filepath_or_buffer,
+    format=None,
+    index=None,
+    encoding=None,
+    chunksize=None,
+    iterator=False,
+    compression="infer",
+):  # pragma: no cover  # noqa: PR01, RT01, D200
```
```suggestion
    format: str | None = None,
    index: Hashable | None = None,
    encoding: str | None = None,
    chunksize: int | None = None,
    iterator: bool = False,
    compression: CompressionOptions = "infer",
) -> DataFrame | ReaderBase:  # noqa: PR01, RT01, D200
```

anmyachev commented on modin/pandas/io.py:
```diff
@@ -93,6 +489,91 @@ def read_hdf(
+@enable_logging
+def read_stata(
+    filepath_or_buffer,
+    convert_dates=True,
+    convert_categoricals=True,
+    index_col=None,
+    convert_missing=False,
+    preserve_dtypes=True,
+    columns=None,
+    order_categoricals=True,
+    chunksize=None,
+    iterator=False,
+    compression="infer",
+    storage_options: "StorageOptions" = None,
+):
```
```suggestion
    convert_dates: bool = True,
    convert_categoricals: bool = True,
    index_col: str | None = None,
    convert_missing: bool = False,
    preserve_dtypes: bool = True,
    columns: Sequence[str] | None = None,
    order_categoricals: bool = True,
    chunksize: int | None = None,
    iterator: bool = False,
    compression: CompressionOptions = "infer",
    storage_options: StorageOptions = None,
) -> DataFrame | StataReader:
```

anmyachev commented on modin/pandas/io.py:
```diff
@@ -93,6 +489,91 @@ def read_hdf(
+@enable_logging
+def read_feather(
+    path,
+    columns=None,
+    use_threads: bool = True,
+    storage_options: "StorageOptions" = None,
```
```suggestion
    columns: Sequence[Hashable] | None = None,
    use_threads: bool = True,
    storage_options: StorageOptions = None,
```

anmyachev commented on modin/pandas/io.py:
```diff
@@ -19,38 +19,385 @@
+@enable_logging
+def read_html(
+    io,
+    match=".+",
+    flavor=None,
+    header=None,
+    index_col=None,
+    skiprows=None,
+    attrs=None,
+    parse_dates=False,
+    thousands=",",
+    encoding=None,
+    decimal=".",
+    converters=None,
+    na_values=None,
+    keep_default_na=True,
+    displayed_only=True,
+    extract_links=None,
+):  # noqa: PR01, RT01, D200
```
```suggestion
    match: str | Pattern = ".+",
    flavor: str | None = None,
    header: int | Sequence[int] | None = None,
    index_col: int | Sequence[int] | None = None,
    skiprows: int | Sequence[int] | slice | None = None,
    attrs: dict[str, str] | None = None,
    parse_dates: bool = False,
    thousands: str | None = ",",
    encoding: str | None = None,
    decimal: str = ".",
    converters: dict | None = None,
    na_values: Iterable[object] | None = None,
    keep_default_na: bool = True,
    displayed_only: bool = True,
    extract_links: Literal[None, "header", "footer", "body", "all"] = None,
) -> list[DataFrame]:  # noqa: PR01, RT01, D200
```

anmyachev commented on modin/pandas/io.py:
```diff
@@ -19,38 +19,385 @@
+@enable_logging
+def read_gbq(
+    query: str,
+    project_id: Optional[str] = None,
+    index_col: Optional[str] = None,
+    col_order: Optional[List[str]] = None,
+    reauth: bool = False,
+    auth_local_webserver: bool = True,
+    dialect: Optional[str] = None,
+    location: Optional[str] = None,
+    configuration: Optional[Dict[str, Any]] = None,
+    credentials=None,
+    use_bqstorage_api: Optional[bool] = None,
+    progress_bar_type: Optional[str] = None,
+    max_results: Optional[int] = None,
```
```suggestion
    project_id: str | None = None,
    index_col: str | None = None,
    col_order: list[str] | None = None,
    reauth: bool = False,
    auth_local_webserver: bool = True,
    dialect: str | None = None,
    location: str | None = None,
    configuration: dict[str, Any] | None = None,
    credentials=None,
    use_bqstorage_api: bool | None = None,
    max_results: int | None = None,
    progress_bar_type: str | None = None,
```

anmyachev commented on modin/pandas/io.py:
```diff
@@ -19,38 +19,385 @@
+@enable_logging
+def read_json(
+    path_or_buf,
+    orient=None,
+    typ="frame",
+    dtype=None,
+    convert_axes=None,
+    convert_dates=True,
+    keep_default_dates=True,
+    numpy=False,
+    precise_float=False,
+    date_unit=None,
+    encoding=None,
+    encoding_errors="strict",
+    lines=False,
+    chunksize=None,
+    compression="infer",
+    nrows: Optional[int] = None,
+    storage_options: "StorageOptions" = None,
+):
```
```suggestion
    orient: str | None = None,
    typ: Literal["frame", "series"] = "frame",
    dtype: DtypeArg | None = None,
    convert_axes=None,
    convert_dates: bool | list[str] = True,
    keep_default_dates: bool = True,
    numpy: bool = False,
    precise_float: bool = False,
    date_unit: str | None = None,
    encoding: str | None = None,
    encoding_errors: str | None = "strict",
    lines: bool = False,
    chunksize: int | None = None,
    compression: CompressionOptions = "infer",
    nrows: int | None = None,
    storage_options: StorageOptions = None,
) -> DataFrame | Series | JsonReader:
```

anmyachev commented on modin/pandas/io.py:
```diff
@@ -19,38 +19,385 @@
+@enable_logging
+def read_parquet(
+    path,
+    engine: str = "auto",
+    columns=None,
+    storage_options: "StorageOptions" = None,
+    use_nullable_dtypes: bool = False,
+    **kwargs,
+):
```
```suggestion
    engine: str = "auto",
    columns: list[str] | None = None,
    storage_options: StorageOptions = None,
    use_nullable_dtypes: bool = False,
    **kwargs,
) -> DataFrame:
```

anmyachev commented on modin/pandas/io.py:
```diff
@@ -19,38 +19,385 @@
+@_inherit_docstrings(pandas.read_table, apilink="pandas.read_table")
+@enable_logging
+def read_table(
+    filepath_or_buffer: "FilePath | ReadCsvBuffer[bytes] | ReadCsvBuffer[str]",
+    sep=no_default,
+    delimiter=None,
+    header="infer",
+    names=no_default,
+    index_col=None,
+    usecols=None,
+    squeeze=None,
+    prefix=no_default,
+    mangle_dupe_cols=True,
+    dtype: "DtypeArg | None" = None,
+    engine: "CSVEngine | None" = None,
+    converters=None,
+    true_values=None,
+    false_values=None,
+    skipinitialspace=False,
+    skiprows=None,
+    skipfooter=0,
```
```suggestion
    filepath_or_buffer: FilePath | ReadCsvBuffer[bytes] | ReadCsvBuffer[str],
    sep: str | None | lib.NoDefault = lib.no_default,
    delimiter: str | None | lib.NoDefault = None,
    # Column and Index Locations and Names
    header: int | Sequence[int] | None | Literal["infer"] = "infer",
    names: Sequence[Hashable] | None | lib.NoDefault = lib.no_default,
    index_col: IndexLabel | Literal[False] | None = None,
    usecols=None,
    squeeze: bool | None = None,
    prefix: str | lib.NoDefault = lib.no_default,
    mangle_dupe_cols: bool = True,
    # General Parsing Configuration
    dtype: DtypeArg | None = None,
    engine: CSVEngine | None = None,
    converters=None,
    true_values=None,
    false_values=None,
    skipinitialspace: bool = False,
    skiprows=None,
    skipfooter: int = 0,
```

anmyachev commented on modin/pandas/io.py:
```diff
@@ -19,38 +19,385 @@
+    skipinitialspace=False,
+    skiprows=None,
+    skipfooter=0,
+    nrows=None,
+    na_values=None,
+    keep_default_na=True,
+    na_filter=True,
+    verbose=False,
+    skip_blank_lines=True,
+    parse_dates=False,
+    infer_datetime_format=False,
+    keep_date_col=False,
+    date_parser=None,
+    dayfirst=False,
+    cache_dates=True,
+    iterator=False,
+    chunksize=None,
+    compression: "CompressionOptions" = "infer",
+    thousands=None,
```
```suggestion
    nrows: int | None = None,
    # NA and Missing Data Handling
    na_values=None,
    keep_default_na: bool = True,
    na_filter: bool = True,
    verbose: bool = False,
    skip_blank_lines: bool = True,
    # Datetime Handling
    parse_dates=False,
    infer_datetime_format: bool = False,
    keep_date_col: bool = False,
    date_parser=None,
    dayfirst: bool = False,
    cache_dates: bool = True,
    # Iteration
    iterator: bool = False,
    chunksize: int | None = None,
    # Quoting, Compression, and File Format
    compression: CompressionOptions = "infer",
    thousands: str | None = None,
```

anmyachev commented on modin/pandas/io.py:
```diff
@@ -19,38 +19,385 @@
+    chunksize=None,
+    compression: "CompressionOptions" = "infer",
+    thousands=None,
+    decimal: "str" = ".",
+    lineterminator=None,
+    quotechar='"',
+    quoting=0,
+    doublequote=True,
+    escapechar=None,
+    comment=None,
+    encoding=None,
+    encoding_errors: "str | None" = "strict",
+    dialect=None,
+    error_bad_lines=None,
+    warn_bad_lines=None,
+    on_bad_lines=None,
+    delim_whitespace=False,
+    low_memory=True,
+    memory_map=False,
```
```suggestion
    decimal: str = ".",
    lineterminator: str | None = None,
    quotechar: str = '"',
    quoting: int = csv.QUOTE_MINIMAL,
    doublequote: bool = True,
    escapechar: str | None = None,
    comment: str | None = None,
    encoding: str | None = None,
    encoding_errors: str | None = "strict",
    dialect: str | csv.Dialect | None = None,
    # Error Handling
    error_bad_lines: bool | None = None,
    warn_bad_lines: bool | None = None,
    on_bad_lines=None,
    # Internal
    delim_whitespace=False,
    low_memory=_c_parser_defaults["low_memory"],
    memory_map: bool = False,
```

anmyachev commented on modin/pandas/io.py:
```diff
@@ -19,38 +19,385 @@
+    delim_whitespace=False,
+    low_memory=True,
+    memory_map=False,
+    float_precision=None,
+    storage_options: "StorageOptions" = None,
+):
```
```suggestion
    float_precision: str | None = None,
    storage_options: StorageOptions = None,
) -> DataFrame | TextFileReader:
```

anmyachev commented on modin/pandas/io.py:
```diff
@@ -19,38 +19,385 @@
+@_inherit_docstrings(pandas.read_csv, apilink="pandas.read_csv")
+@enable_logging
+def read_csv(
+    filepath_or_buffer: "FilePath | ReadCsvBuffer[bytes] | ReadCsvBuffer[str]",
+    sep=no_default,
+    delimiter=None,
+    header="infer",
+    names=no_default,
+    index_col=None,
+    usecols=None,
+    squeeze=None,
+    prefix=no_default,
+    mangle_dupe_cols=True,
+    dtype: "DtypeArg | None" = None,
+    engine: "CSVEngine | None" = None,
+    converters=None,
+    true_values=None,
+    false_values=None,
+    skipinitialspace=False,
+    skiprows=None,
+    nrows=None,
```
```suggestion
    filepath_or_buffer: FilePath | ReadCsvBuffer[bytes] | ReadCsvBuffer[str],
    sep: str | None | lib.NoDefault = lib.no_default,
    delimiter: str | None | lib.NoDefault = None,
    # Column and Index Locations and Names
    header: int | Sequence[int] | None | Literal["infer"] = "infer",
    names: Sequence[Hashable] | None | lib.NoDefault = lib.no_default,
    index_col: IndexLabel | Literal[False] | None = None,
    usecols=None,
    squeeze: bool | None = None,
    prefix: str | lib.NoDefault = lib.no_default,
    mangle_dupe_cols: bool = True,
    # General Parsing Configuration
    dtype: DtypeArg | None = None,
    engine: CSVEngine | None = None,
    converters=None,
    true_values=None,
    false_values=None,
    skipinitialspace: bool = False,
    skiprows=None,
    skipfooter: int = 0,
    nrows: int | None = None,
```

anmyachev commented on modin/pandas/io.py:
```diff
@@ -19,38 +19,385 @@
+    skipinitialspace=False,
+    skiprows=None,
+    nrows=None,
+    na_values=None,
+    keep_default_na=True,
+    na_filter=True,
+    verbose=False,
+    skip_blank_lines=True,
+    parse_dates=None,
+    infer_datetime_format=False,
+    keep_date_col=False,
+    date_parser=None,
+    dayfirst=False,
+    cache_dates=True,
+    iterator=False,
+    chunksize=None,
+    compression: "CompressionOptions" = "infer",
+    thousands=None,
+    decimal: "str" = ".",
+    lineterminator=None,
+    quotechar='"',
```
```suggestion
    # NA and Missing Data Handling
    na_values=None,
    keep_default_na: bool = True,
    na_filter: bool = True,
    verbose: bool = False,
    skip_blank_lines: bool = True,
    # Datetime Handling
    parse_dates=None,
    infer_datetime_format: bool = False,
    keep_date_col: bool = False,
    date_parser=None,
    dayfirst: bool = False,
    cache_dates: bool = True,
    # Iteration
    iterator: bool = False,
    chunksize: int | None = None,
    # Quoting, Compression, and File Format
    compression: CompressionOptions = "infer",
    thousands: str | None = None,
    decimal: str = ".",
    lineterminator: str | None = None,
    quotechar: str = '"',
```

anmyachev commented on modin/pandas/io.py:
```diff
@@ -19,38 +19,385 @@
+    decimal: "str" = ".",
+    lineterminator=None,
+    quotechar='"',
+    quoting=0,
+    escapechar=None,
+    comment=None,
+    encoding=None,
+    encoding_errors: "str | None" = "strict",
+    dialect=None,
+    error_bad_lines=None,
+    warn_bad_lines=None,
+    on_bad_lines=None,
+    skipfooter=0,
+    doublequote=True,
+    delim_whitespace=False,
+    low_memory=True,
+    memory_map=False,
+    float_precision=None,
+    storage_options: "StorageOptions" = None,
+):
```
```suggestion
    quoting: int = csv.QUOTE_MINIMAL,
    doublequote: bool = True,
    escapechar: str | None = None,
    comment: str | None = None,
    encoding: str | None = None,
    encoding_errors: str | None = "strict",
    dialect: str | csv.Dialect | None = None,
    # Error Handling
    error_bad_lines: bool | None = None,
    warn_bad_lines: bool | None = None,
    on_bad_lines=None,
    # Internal
    delim_whitespace: bool = False,
    low_memory=_c_parser_defaults["low_memory"],
    memory_map: bool = False,
    float_precision: Literal["high", "legacy"] | None = None,
    storage_options: StorageOptions = None,
) -> DataFrame | TextFileReader:
```

anmyachev commented on modin/pandas/io.py:
```diff
@@ -19,38 +19,385 @@
+@_inherit_docstrings(pandas.read_xml, apilink="pandas.read_xml")
+@enable_logging
+def read_xml(
+    path_or_buffer,
+    xpath="./*",
+    namespaces=None,
+    elems_only=False,
+    attrs_only=False,
+    names=None,
+    dtype=None,
+    converters=None,
+    parse_dates=None,
+    encoding="utf-8",
+    parser="lxml",
+    stylesheet=None,
+    iterparse=None,
+    compression="infer",
+    storage_options=None,
```
```suggestion
    path_or_buffer: FilePath | ReadBuffer[bytes] | ReadBuffer[str],
    xpath: str = "./*",
    namespaces: dict[str, str] | None = None,
    elems_only: bool = False,
    attrs_only: bool = False,
    names: Sequence[str] | None = None,
    dtype: DtypeArg | None = None,
    converters: ConvertersArg | None = None,
    parse_dates: ParseDatesArg | None = None,
    encoding: str | None = "utf-8",
    parser: XMLParsers = "lxml",
    stylesheet: FilePath | ReadBuffer[bytes] | ReadBuffer[str] | None = None,
    iterparse: dict[str, list[str]] | None = None,
    compression: CompressionOptions = "infer",
    storage_options: StorageOptions = None,
```

YarShev commented:
@mvashishtha, `Code scanning results / CodeQL` failed, please take a look.

anmyachev commented on modin/pandas/resample.py:
```diff
@@ -17,20 +17,78 @@
+        """
+        df = self._dataframe if self.axis == 0 else self._dataframe.T
+        groups = df.groupby(
+            pandas.Grouper(
+                key=self.resample_kwargs["on"],
+                freq=self.resample_kwargs["rule"],
+                closed=self.resample_kwargs["closed"],
+                label=self.resample_kwargs["label"],
+                convention=self.resample_kwargs["convention"],
+                loffset=self.resample_kwargs["loffset"],
+                base=self.resample_kwargs["base"],
+                level=self.resample_kwargs["level"],
+                origin=self.resample_kwargs["origin"],
+                offset=self.resample_kwargs["offset"],
+            ),
```
looks like a bug

anmyachev commented:
> @mvashishtha, `Code scanning results / CodeQL` failed, please take a look.

One of the problem fixed here https://github.com/modin-project/modin/pull/5288

mvashishtha commented:
@anmyachev @YarShev for the remaining CodeQL errors:

- [this one](https://github.com/modin-project/modin/security/code-scanning/350) @anmyachev will fix [here](https://github.com/modin-project/modin/pull/5288)
- [this one](https://github.com/modin-project/modin/security/code-scanning/376) existed before we added python 3.6 support and I filed #5294  for it
- [this one](https://github.com/modin-project/modin/security/code-scanning/377) existed before we added python 3.6 support and I filed #5295 for it
- [this one](https://github.com/modin-project/modin/security/code-scanning/393) is incorrect as @anmyachev pointed out, so I closed it
- [this one](https://github.com/modin-project/modin/security/code-scanning/395) is legitimate and I fixed it with my latest commit.

anmyachev reviewed (dismissed):
LGTM! We can ignore `CodeQL` check.

YarShev commented:
@mvashishtha, LGTM, thanks! One comment from my side.

We should fix `setup.py`.

mvashishtha commented:
> We should fix setup.py.

@YarShev Done.

YarShev commented:
@mvashishtha, we should revert all the changes in `setup.py` introduced in https://github.com/modin-project/modin/commit/6ce9cf4daec7f9996038205289bce2186be87611.

mvashishtha commented:
> we should revert all the changes in setup.py introduced in https://github.com/modin-project/modin/commit/6ce9cf4daec7f9996038205289bce2186be87611.

@YarShev done.

YarShev commented on setup.py:
```diff
@@ -56,7 +50,13 @@ def make_distribution(self):
     long_description_content_type="text/markdown",
-    install_requires=[f"pandas=={PANDAS_VERSION}", "packaging", "numpy>=1.18.5", "fsspec", "psutil"],
+    install_requires=[
+        f"pandas==1.5.2",
```
```suggestion
        "pandas==1.5.2",
```

anmyachev reviewed (approved):
LGTM!
